presto-main/src/main/java/com/facebook/presto/operator/aggregation/AccumulatorCompiler.java

```diff
@@ -543,14 +545,32 @@ private static BytecodeBlock generateInputForLoop(
                         .invokeStatic(CompilerOperations.class, "testMask", boolean.class, Block.class, int.class))
                 .ifTrue(loopBody);
 
-        block.append(new ForLoop()
+        BytecodeNode forLoop = new ForLoop()
                 .initialize(new BytecodeBlock().putVariable(positionVariable, 0))
                 .condition(new BytecodeBlock()
                         .getVariable(positionVariable)
                         .getVariable(rowsVariable)
                         .invokeStatic(CompilerOperations.class, "lessThan", boolean.class, int.class, int.class))
                 .update(new BytecodeBlock().incrementVariable(positionVariable, (byte) 1))
-                .body(loopBody));
+                .body(loopBody);
+
+        for (int i = 0; i < parameterVariables.size(); i++) {
+            if (!nullable.get(i)) {
+                Variable variableDefinition = parameterVariables.get(i);
+                forLoop = new IfStatement("if(!(%s instanceof RunLengthEncodedBlock && %s.isNull(0)))", variableDefinition.getName(), variableDefinition.getName())
```

Comment:
nit: what about `if (!(%s ...` (basically add a space

Reply:
@wenleix I was matching other comments in this file. They don't have space.

```
loopBody = new IfStatement("if(testMask(%s, position))", masksBlock.getName())

loopBody = new IfStatement("if(!%s.isNull(position))", variableDefinition.getName())
```